tuna/miopen/driver/convolution.py

```diff
@@ -125,7 +134,7 @@ def parse_fdb_key(self, line):
     if search(pattern_3d, line):
       setattr(self, 'spatial_dim', 3)
 
-  def parse_driver_line(self, line):
+  def parse_driver_line(self, line: str) -> None:
     super().parse_driver_line(line)
 
     if self.direction and self.direction in DIRECTION:
```

Comment:
Technically self.direction is initialized as an int in the set (1,2,4) by parse_driver_line. This line (140) resolves it to a string.